src/hooks/useFirestore.js

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react';
+import { db } from '../lib/firebase';
+
+const useFirestore = (collection) => {
+  const [docs, setDocs] = useState([]);
+
+  useEffect(() => {
+    const unsubscribe = db.collection(collection).onSnapshot((snapshot) => {
+      let documents = [];
```

Comment:
Fun fact - you could actually still use a `const` here, even though you're pushing new items onto the array. Const just makes the _reference_ or variable name immutable, but not the data that the reference/name points to.

So like `documents = someOtherList` would throw an error with `const` but `documents.push({}` would not.

Reply:
Ah, good to know! I'll make that change before merging.